Distributedlog Core Shaded Jar fails with a NoClassDefFoundError

**BUG REPORT**

***Describe the bug***
After upgrading to a new version of `distributedlog-core-shaded` jar in Apache Heron, I ran into the following error message. It seems that some of the transitive dependencies are not being included in the shaded jar.

```
Caused by: org.glassfish.jersey.server.ContainerException: java.lang.NoClassDefFoundError: dlshade/com/google/common/util/concurrent/internal/InternalFutureFailureAccess
	at org.glassfish.jersey.servlet.internal.ResponseWriter.rethrow(ResponseWriter.java:278)
	at org.glassfish.jersey.servlet.internal.ResponseWriter.failure(ResponseWriter.java:260)
	at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:509)
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:334)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154)
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:473)
	... 27 more
```

This is the list of transitive dependencies that are not being included. Is this by design? It seems that by not including the dependencies, but still shading all of `com.google` to `dlshade.com.google` there are bugs with the generated shaded jar.
```
❯ mvn install | grep -i excluding
[INFO] Excluding com.google.guava:failureaccess:jar:1.0.1 from the shaded jar.
[INFO] Excluding com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava from the shaded jar.
[INFO] Excluding org.checkerframework:checker-qual:jar:3.5.0 from the shaded jar.
[INFO] Excluding com.google.errorprone:error_prone_annotations:jar:2.4.0 from the shaded jar.
[INFO] Excluding com.google.j2objc:j2objc-annotations:jar:1.3 from the shaded jar.
[INFO] Excluding org.apache.yetus:audience-annotations:jar:0.5.0 from the shaded jar.
[INFO] Excluding org.jctools:jctools-core:jar:2.1.2 from the shaded jar.
[INFO] Excluding io.netty:netty-handler:jar:4.1.50.Final from the shaded jar.
[INFO] Excluding io.netty:netty-resolver:jar:4.1.50.Final from the shaded jar.
[INFO] Excluding io.netty:netty-transport:jar:4.1.50.Final from the shaded jar.
[INFO] Excluding io.netty:netty-codec:jar:4.1.50.Final from the shaded jar.
[INFO] Excluding io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.50.Final from the shaded jar.
[INFO] Excluding io.netty:netty-transport-native-unix-common:jar:4.1.50.Final from the shaded jar.
[INFO] Excluding io.netty:netty-tcnative-boringssl-static:jar:2.0.31.Final from the shaded jar.
[INFO] Excluding org.bouncycastle:bcpkix-jdk15on:jar:1.60 from the shaded jar.
[INFO] Excluding org.bouncycastle:bcprov-jdk15on:jar:1.60 from the shaded jar.
[INFO] Excluding org.bouncycastle:bcprov-ext-jdk15on:jar:1.60 from the shaded jar.
[INFO] Excluding com.beust:jcommander:jar:1.78 from the shaded jar.
[INFO] Excluding com.google.code.findbugs:jsr305:jar:3.0.2 from the shaded jar.
[INFO] Excluding org.slf4j:slf4j-api:jar:1.7.25 from the shaded jar.
[INFO] Excluding commons-configuration:commons-configuration:jar:1.10 from the shaded jar.
```

--- a/shaded/distributedlog-core-shaded/pom.xml
+++ b/shaded/distributedlog-core-shaded/pom.xml
@@ -36,20 +36,8 @@
       <version>${project.version}</version>
       <exclusions>
         <exclusion>
-          <groupId>org.slf4j</groupId>
-          <artifactId>slf4j-log4j12</artifactId>
-        </exclusion>
-        <exclusion>
-          <groupId>org.slf4j</groupId>
-          <artifactId>slf4j-reload4j</artifactId>
-        </exclusion>
-        <exclusion>
           <groupId>log4j</groupId>
           <artifactId>log4j</artifactId>
-        </exclusion>
-        <exclusion>
-          <groupId>ch.qos.reload4j</groupId>
-          <artifactId>reload4j</artifactId>
         </exclusion>
         <exclusion>
           <groupId>io.netty</groupId>
@@ -83,6 +71,7 @@
                   <include>com.fasterxml.jackson.core:jackson-core</include>
                   <include>com.fasterxml.jackson.core:jackson-databind</include>
                   <include>com.fasterxml.jackson.core:jackson-annotations</include>
+                  <include>com.google.guava:failureaccess</include>
                   <include>com.google.guava:guava</include>
                   <include>com.google.protobuf:protobuf-java</include>
                   <include>net.jpountz.lz4:lz4</include>

